src/box/lua/config/applier/box_cfg.lua

```diff
@@ -283,7 +283,9 @@ local function names_apply(config, missing_names, schema_version)
     names_alert_missing(config, missing_names)
     -- Don't wait for box.status to change, we may be already rw, set names
     -- on reload, if it's possible and needed.
-    names_try_set_missing()
+    if schema_version and schema_version >= mkversion(3, 0, 0) then
+        names_try_set_missing()
+    end
```

Comment:
Will this check fail on tarantool 3.1?

Shoudn't we check that the schema is the last one, not just >= 3.0.0?

Reply:
Agree, I propose to export `box_latest_dd_version_id` the same way it's done with `box_dd_version_id` and add this to `mkversion.lua` file. This way we'll get latest schema version

Reply:
Good catch, will do.

Reply:
Samee stands for https://github.com/tarantool/tarantool/blob/26acba832c4e33668b6eef52051a7d1f4a30ee2e/src/box/lua/config/applier/box_cfg.lua#L298-L300
and
https://github.com/tarantool/tarantool/blob/26acba832c4e33668b6eef52051a7d1f4a30ee2e/src/box/lua/config/applier/box_cfg.lua#L213-L223
right?

Reply:
Right, we're not supposed to set names before the time, the schema is latest. We cannot do it at all. Missed that during initial implementation

Reply:
OTOH, it works while the latest schema is 3.0.0 and we can rework it after the 3.0.0 release.

Reply:
> OTOH, it works while the latest schema is 3.0.0 and we can rework it after the 3.0.0 release.

It was quite easy to fix so I did this as well.

Reply:
Thank you!